src/braille/internal/braille.cpp

```diff
@@ -310,6 +316,292 @@ namespace mu::engraving {
+    case 23456: return "⠾";
+    case 123456: return "⠿";
+    }
+    return " ";
+}
+
+std::string translate2Braille(std::string codes)
+{
+    std::stringstream test(codes);
+    std::string segment;
+    std::vector<std::string> seglist;
+
+    std::string txt = "";
+    while (std::getline(test, segment, '-')) {
+        txt.append(getBraillePattern(segment));
+    }
+    return txt;
+}
+
+std::string intToBrailleUpperNumbers(std::string txt, bool indicator)
+{
+    std::string braille = "";
+    if (indicator) {
+        braille.append(translate2Braille(Braille_NumIndicator));
+    }
+
+    for (size_t i=0; i < txt.length(); i++) {
+        char c = txt.at(i);
+        if (c - '0' <= 9) {
+            braille.append(translate2Braille(Braille_UpperNumbers[c - '0']));
+        }
+    }
+    return braille;
+}
+
+std::string intToBrailleLowerNumbers(std::string txt, bool indicator)
+{
+    std::string braille = "";
+    if (indicator) {
+        braille.append(Braille_NumIndicator);
+    }
+
+    for (size_t i=0; i < txt.length(); i++) {
+        char c = txt.at(i);
+        if (c - '0' <= 9) {
+            braille.append(Braille_LowerNumbers[c - '0']);
+        }
+    }
+    return braille;
+}
+
+BrailleEngravingItems::BrailleEngravingItems()
+{
+    m_braille_str = QString();
+}
+
+BrailleEngravingItems::~BrailleEngravingItems()
+{
+    clear();
+}
+
+void BrailleEngravingItems::clear()
+{
+    m_braille_str = QString();
+    m_items.clear();
+}
+
+void BrailleEngravingItems::join(BrailleEngravingItems* another, bool newline, bool del)
+{
+    int len = m_braille_str.length();
+
+    if (newline && !m_braille_str.isEmpty()) {
+        m_braille_str.append("\n");
+        len++;
+    }
+
+    m_braille_str.append(another->brailleStr());
+
+    for (auto item: *another->items()) {
+        item.second.first += len;
+        item.second.second += len;
+        m_items.push_back(item);
+    }
+    if (del) {
+        delete another;
+    }
+}
+
+void BrailleEngravingItems::join(std::vector<BrailleEngravingItems*> lst, bool newline, bool del)
+{
+    for (auto item: lst) {
+        join(item, newline, del);
+    }
+}
+
+QString BrailleEngravingItems::brailleStr()
+{
+    return m_braille_str;
+}
+
+std::vector<std::pair<EngravingItem*, std::pair<int, int> > >* BrailleEngravingItems::items()
+{
+    return &m_items;
+}
+
+void BrailleEngravingItems::setBrailleStr(QString str)
+{
+    m_braille_str = str;
+    m_items.clear();
+}
+
+void BrailleEngravingItems::addPrefixStr(QString str)
+{
+    int len = str.length();
+    m_braille_str = str.append(m_braille_str);
+
+    for (size_t i=0; i < m_items.size(); i++) {
+        m_items[i].second.first += len;
+        m_items[i].second.second += len;
+    }
+}
+
+void BrailleEngravingItems::addEngravingItem(EngravingItem* el, QString braille)
+{
+    //braille = braille.replace(QRegularExpression ("/\\/"), "\\\\");
+    //Manual doubling slashes '\' because Regex doesn't work. Don't know why.
```

Comment:
Wouldn't you need to write `\\\\` to match a single backslash? Namely one  `\\` to escape it for C++ (which puts a single backslash in your regex), and then another `\\` to make an escaped backslash in your regex. So, QRegularExpression("\\\\") matches a single backslash. (See also the examples in https://doc.qt.io/qt-5/qregularexpression.html#introduction.)

Also, are the forward slashes part of the pattern? (I'm wondering because in JS for example, forward slashes are used as delimiters of the pattern, but in the case of `QRegularExpression`, such delimiters shouldn't be included.)

And why do you need to use regex? Doesn't the overload that takes simply two QStrings do the job?